File: pyround1/455.分发饼干.py
#
# @lc app=leetcode.cn id=455 lang=python
#
# [455] 分发饼干
#

# @lc code=start
class Solution(object):
    def findContentChildren(self, g, s):
        """
        :type g: List[int]
        :type s: List[int]
        :rtype: int
        """
        s.sort()
        g.sort()
        child, cookie = 0, 0
        while child < len(g) and cookie < len(s):
            if g[child] <= s[cookie]:
                child += 1
            cookie += 1
        
        return child
        
# @lc code=end

File: 455.分发饼干.cpp
/*
 * @lc app=leetcode.cn id=455 lang=cpp
 *
 * [455] 分发饼干
 */

// @lc code=start
class Solution {
public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        int count = 0;
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int j = 0;
        for(int i = 0; i < g.size(); i++){
            while(j < s.size() and g[i] > s[j])
            {
                j++;
            }
            if(i > s.size() or j >= s.size())
                break;
            count++;
            j++;
        }
        return count;
    }
};
// @lc code=end

